orchestrator/services/subscriptions.py

```diff
@@ -409,12 +418,21 @@ def query_depends_on_subscriptions(subscription_id: UUID) -> Query:
         .with_entities(SubscriptionTable.subscription_id)
     )
 
-    return SubscriptionTable.query.filter(
-        or_(
-            SubscriptionTable.subscription_id.in_(resource_type_relations.scalar_subquery()),
-            SubscriptionTable.subscription_id.in_(relation_relations.scalar_subquery()),
+    if filter_statuses:
```

Comment:
adding all filter_statuses to the variable would probably look a bit better then two queries almost exactly the same with an if statement.